addons/product/models/product_pricelist.py

```diff
@@ -593,6 +559,59 @@ def write(self, values):
         self.invalidate_cache()
         return res
 
+    def _get_first_applicable_rule(self, product, qty, uom):
+        """Return the first pricelist rule applicable to product & qty.
+
+        :param product: product record (product.product/product.template)
+        :param float qty: quantity of products requested (in given uom)
+        :param uom: unit of measure (uom.uom record)
+        :returns: first pricelist rule found matching qty & product/template/category id
+        :rtype: product.pricelist.item record
+        """
+        product.ensure_one()
+        is_product_template = product._name == "product.template"
+        suitable_rule = self.browse()
+
+        # Compute quantity in product uom because pricelist rules are specified
+        # w.r.t product default UoM (min_quantity, price_surchage, ...)
+        product_uom = product.uom_id
+        if uom != product_uom:
+            qty_in_product_uom = uom._compute_quantity(qty, product_uom, raise_if_failure=False)
+        else:
+            qty_in_product_uom = qty
+
+        for rule in self:
+            if rule.min_quantity and qty_in_product_uom < rule.min_quantity:
+                continue
+
```

Comment:
After more thoughts on it, we should instead follow the idea of https://github.com/odoo/odoo/pull/81350 and keep the loop & qty_in_product_uom computation in `_compute_price_rule`